worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/WorldEditListener.java

```diff
@@ -86,6 +93,19 @@ public void onPlayerCommandSend(PlayerCommandSendEvent event) {
         );
     }
 
+    private static int getCurrentTick() {
+        if (PaperLib.isPaper()) {
+            return Bukkit.getCurrentTick();
+        }
+        return (int) (System.currentTimeMillis() / 50);
```

Comment:
It is not truly arbitrary. Since Spigot does not provide a way to get the current tick, I use the current time as an approximation. There is an offset between the real current tick and the approximation but we only want to detect when the delay between two clicks is too short so the offset is not a problem. Sure we can use `nanoTime`, this would improve accuracy a bit. I doubt it will have a significant impact but it cannot hurt.

Reply:
I mainly want to avoid time going backwards, which could cause oddities.

Reply:
Oh yes, I forgot that `currentTimeMillis` is not monotonic, while `nanoTime` is.